python/benchmarks/bi_benchmarks.py

```diff
@@ -0,0 +1,149 @@
+    # We use dataframe in this file
+    CITY_BI_FILE = "data/CityMaxCapita_1.csv.bz2"
+    CITY_BI_FILE2 = "data/CityMaxCapita_1.parquet.gzip"
+
+    #Defines how many times bigger the database is
+    params = [1, 10]
+
+    def __init__(self):
+        self.lib_name = BIBenchmarks.LIB_NAME
+        self.symbol = self.lib_name
+
+    def setup_cache(self):
+
+        file = os.path.join(Path(__file__).resolve().parent.parent, BIBenchmarks.CITY_BI_FILE2)
+        if (not os.path.exists(file)) :
+            dfo = download_and_process_city_to_parquet(file)
+            dff = pd.read_parquet(file)
+            pd.testing.assert_frame_equal(dfo,dff)
+        else:
+            print("Parquet file exists!")
+
+        # read data from bz.2 file
+        # abs_path = os.path.join(Path(__file__).resolve().parent.parent,BIBenchmarks.CITY_BI_FILE)
+        # self.df : pd.DataFrame = process_city(abs_path)
+
+        self.df : pd.DataFrame = pd.read_parquet(file)
+
+        self.ac = Arctic(f"lmdb://opensource_datasets_{self.lib_name}?map_size=20GB")
+        self.ac.delete_library(self.lib_name)
+        self.lib = self.ac.create_library(self.lib_name)
+
+        print("The procedure is creating N times larger dataframes")
+        print("by concatenating original DF N times")
+        for num in BIBenchmarks.params:
+            _df = self.df.copy(deep=True)
+            if (num > 1):
+                # lets create N times bigger DF
+                dfcum = self.df.copy(deep=True)
+                for i in range(1, (BIBenchmarks.params[-1])):
+                    dfcum = pd.concat([dfcum, self.df])            
+                _df = dfcum
+            self.lib.write(f"{self.symbol}{num}", _df)
+
+        print("If pandas query produces different dataframe than arctic one stop tests!")
+        print("This will mean query problem is there most likely")
+
+        print("Pre-check correctness for query_groupaby_city_count_all")
+        _df = self.df.copy(deep=True)
+        arctic_df = self.time_query_groupaby_city_count_all(BIBenchmarks.params[0])
+        _df = self.get_query_groupaby_city_count_all(_df)
+        arctic_df.sort_index(inplace=True)
+        self.assert_frame_equal(_df, arctic_df)
+
+        print("Pre-check correctness for query_groupaby_city_count_isin_filter")
+        _df = self.df.copy(deep=True)
+        arctic_df = self.time_query_groupaby_city_count_isin_filter(BIBenchmarks.params[0])
+        _df = self.get_query_groupaby_city_count_isin_filter(_df)
+        arctic_df.sort_index(inplace=True)
+        self.assert_frame_equal(_df, arctic_df)
+
+        print("Pre-check correctness for query_groupaby_city_count_filter_two_aggregations")
+        _df = self.df.copy(deep=True)
+        arctic_df = self.time_query_groupaby_city_count_filter_two_aggregations(BIBenchmarks.params[0])
+        _df = self.get_query_groupaby_city_count_filter_two_aggregations(_df)
+        arctic_df.sort_index(inplace=True)
```

Comment:
Looks like the sorting of the index is already done in the `assert_frame_equal` function, so it is not needed here.

Reply:
ìndeed! late refactoring omission